From 654af42b3ac991ddefa00979fe3f0dc7aae21965 Mon Sep 17 00:00:00 2001
From: Victor Wan <victor.wan@amlogic.com>
Date: Sat, 17 May 2014 16:41:24 +0800
Subject: [PATCH 4126/5965] meson8/8b: fix array overflow issue

fix compile warning:

arch/arm/mach-meson8b/cpu.c:40:19: warning: array subscript is above array bounds [-Warray-bounds]
arch/arm/mach-meson8b/cpu.c:52:8: warning: array subscript is above array bounds [-Warray-bounds]
---
 arch/arm/mach-meson8/cpu.c  | 4 ++--
 arch/arm/mach-meson8b/cpu.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-meson8/cpu.c b/arch/arm/mach-meson8/cpu.c
index c5487fa08e8e..58ec15da5d70 100755
--- a/arch/arm/mach-meson8/cpu.c
+++ b/arch/arm/mach-meson8/cpu.c
@@ -27,7 +27,7 @@
 #include <linux/printk.h>
 #include <linux/string.h>
 
-static int meson_cpu_version[MESON_CPU_VERSION_LVL_MAX];
+static int meson_cpu_version[MESON_CPU_VERSION_LVL_MAX+1];
 int __init meson_cpu_version_init(void)
 {
 	unsigned int version,ver;	
@@ -71,4 +71,4 @@ int get_meson_cpu_version(int level)
 		return meson_cpu_version[level];
 	return 0;
 }
-EXPORT_SYMBOL(get_meson_cpu_version);
\ No newline at end of file
+EXPORT_SYMBOL(get_meson_cpu_version);
diff --git a/arch/arm/mach-meson8b/cpu.c b/arch/arm/mach-meson8b/cpu.c
index b733a06a2c69..6f0cc61b8263 100755
--- a/arch/arm/mach-meson8b/cpu.c
+++ b/arch/arm/mach-meson8b/cpu.c
@@ -27,7 +27,7 @@
 #include <linux/printk.h>
 #include <linux/string.h>
 
-static int meson_cpu_version[MESON_CPU_VERSION_LVL_MAX];
+static int meson_cpu_version[MESON_CPU_VERSION_LVL_MAX+1];
 int __init meson_cpu_version_init(void)
 {
 	unsigned int version,ver;
-- 
2.19.0

